crates/payload/builder/src/error.rs

```diff
@@ -22,6 +26,10 @@ pub enum PayloadBuilderError {
     /// Thrown if the payload requests withdrawals before Shanghai activation.
     #[error("withdrawals set before Shanghai activation")]
     WithdrawalsBeforeShanghai,
+    /// Thrown when a deposit transaction fails to convert to a [TransactionSignedEcRecovered].
```

Comment:
nit, you can do
```suggestion
    /// Thrown when a deposit transaction fails to convert to a [reth_primitives::TransactionSignedEcRecovered].
```
or
```suggestion
    /// Thrown when a deposit transaction fails to convert to a [TransactionSignedEcRecovered](reth_primitives::TransactionSignedEcRecovered).
```
and get rid of unused import

Reply:
Curious: does this change doc rendering?